File: 4252-first-unique-even-element/first-unique-even-element.cpp
class Solution {
public:
    int firstUniqueEven(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }
        int element=-1;
        for(int i=0;i<nums.size();i++){
            if(nums[i]%2==0 && mp[nums[i]]==1){
                element=nums[i];
                break;
            }
        }
        return element;
    }
};

File: 4252-first-unique-even-element/first-unique-even-element.java
class Solution {
    public int firstUniqueEven(int[] nums) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
        }
        int element=-1;
        for(int i=0;i<nums.length;i++){
            if(nums[i]%2==0 && mp.get(nums[i])==1){
                element=nums[i];
                break;
            }
        }
        return element;
    }
}